File: tests/test_mergesort.py
from random import randint

from sys import path
path.append('..')

from mergesort import mergesort

def test_mergesort():
    test_list = [randint(1, 50) for i in range(15)]
    test_sorted = test_list[:]
    mergesort(test_sorted, 0, 14)
    if test_sorted == sorted(test_list):
        print("test passes")
        print(test_list)
        print(test_sorted)
    else:
        print("test fails")
        print(test_list)
        print(test_sorted)

if __name__ == "__main__":
    test_mergesort()

File: tests/TestMergesort.java
package Standard_Algorithms.tests;

import Standard_Algorithms.Mergesort;
import java.util.Arrays;

public class TestMergesort
{
    public static void main(String[] args)
    {
        float[] testArray = new float[6];
        for (int i=0; i<6; i++)
            {
                testArray[i] = (float)(20*Math.random());
            }

        float[] testSorted = Arrays.copyOf(testArray, 6);
        Mergesort.mergesort(testSorted, 0, 5);
        
        float[] testLibrarySorted = Arrays.copyOf(testArray, 6);
        Arrays.sort(testLibrarySorted);

        if (Arrays.equals(testSorted, testLibrarySorted))
            {
                System.out.println("test passes");
                System.out.println(Arrays.toString(testArray));
                System.out.println(Arrays.toString(testSorted));
            }
        else
            {
                System.out.println("test fails");
                System.out.println(Arrays.toString(testArray));
                System.out.println(Arrays.toString(testSorted));
            }
    }

}